examples/checkout.c

```diff
@@ -265,12 +267,18 @@ int lg2_checkout(git_repository *repo, int argc, char **argv)
 
 	if (match_arg_separator(&args)) {
 		/**
-		 * Try to checkout the given path
+		 * Try to checkout the given path(s)
 		 */
 
-		fprintf(stderr, "unhandled path-based checkout\n");
-		err = 1;
-		goto cleanup;
+		git_checkout_options copts = GIT_CHECKOUT_OPTIONS_INIT;
+		copts.checkout_strategy = GIT_CHECKOUT_FORCE;
+
+		paths.count = args.argc - args.pos;
+		paths.strings = &args.argv[args.pos];
+		copts.paths = paths;
+
+		err = git_checkout_head(repo, &copts);
```

Comment:
We should print an error message if the function failed.

Reply:
Added printing error if `err != 0`. Not sure how to provoke an error in order to test it though.